### ISSUE TITLE
[Feature]: Styled components bump version

### ISSUE THREAD

**Issue by KelvinOm** [appsmithorg/appsmith#25174]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Updating styled components version to 6

### Why should this be worked on?

1 . Updating styled components version is necessary in order to be able to use container query. 
2. Keep one of our core libraries up-to-date

**Comment by KelvinOm** [comment]
After updating the version in the current code base, we get **116 errors in 73 files**, if in addition to this add library to resolutions ([recommended by the documentation](https://styled-components.com/docs/basics#installation), there will be only 1 instance of library) we will get  **693 errors in 160 files**. If we do not add the library to the resolution, then we will not be able to create client build because error `heap out of memory`.

In summary, I can say that such changes will need significant effort and refactoring of most of the app.

**Comment by KelvinOm** [comment]
Work incorrectly with jest
https://github.com/styled-components/styled-components/issues/4094
https://github.com/styled-components/styled-components/issues/4081

**Comment by KelvinOm** [comment]
The task is still blocked, we are waiting for a decision on the technology to use.

**Comment by KelvinOm** [comment]
The task is still blocked, we are waiting for a decision on the technology to use.

**Comment by KelvinOm** [comment]
@somangshu As a WDS team, we are considering using CSS modules for our components and we have described why. I would like us to start using CSS modules for the main codebase as well. But for this, a decision must be made.

**Comment by KelvinOm** [comment]
We decided to use CSS modules as the main technology for styles. So there are no plans to update styled components. The motivation is [here](https://www.notion.so/appsmith/Story-about-CSS-361c908768b14d28afe322f3d6fe163c).
